terraform/modules/spack/monitoring_db.tf

```diff
@@ -0,0 +1,440 @@
+  repl_instance_allocated_storage            = 64
+  repl_instance_auto_minor_version_upgrade   = true
+  repl_instance_allow_major_version_upgrade  = true
+  repl_instance_apply_immediately            = true
+  repl_instance_engine_version               = "3.4.7" # Must use 3.4.7 to support Postgres 14.x
+  repl_instance_multi_az                     = true
+  repl_instance_preferred_maintenance_window = "sun:10:30-sun:14:30"
+  repl_instance_publicly_accessible          = false
+  repl_instance_class                        = "dms.t3.large"
+
+  # Endpoints
+  endpoints = {
+    source = {
+      # Required
+      endpoint_id   = "gitlab-db-source-endpoint"
+      endpoint_type = "source"
+      engine_name   = "postgres"
+      # Optional
+      database_name = module.gitlab_db.db_instance_name
+      username      = module.gitlab_db.db_instance_username
+      password      = local.gitlab_db_master_password
+      port          = module.gitlab_db.db_instance_port
+      server_name   = module.gitlab_db.db_instance_address
+      tags          = { EndpointType = "source" }
+    }
+
+    destination = {
+      # Note: It's difficult to encode this into terraform at the moment (https://github.com/hashicorp/terraform-provider-aws/issues/23225),
+      # but the following setting needs to be entered into the "Endpoint Settings" of the target endpoint
+      # `AfterConnectScript="SET session_replication_role=replica"`
+
+      # This can also be done with the following command from the CLI:
+      # aws dms modify-endpoint --endpoint-arn 'ENDPOINT_ARN' --postgre-sql-settings='AfterConnectScript="SET session_replication_role=replica"'
+
+      # Required
+      endpoint_id   = "gitlab-db-target-endpoint"
+      endpoint_type = "target"
+      engine_name   = "postgres"
+      # Optional
+      database_name = module.gitlab_db_clone[0].db_instance_name
+      username      = module.gitlab_db_clone[0].db_instance_username
+      password      = local.gitlab_db_clone_master_password
+      port          = module.gitlab_db_clone[0].db_instance_port
+      server_name   = module.gitlab_db_clone[0].db_instance_address
+      tags          = { EndpointType = "destination" }
+    }
+  }
+
+  replication_tasks = {
+    cdc_ex = {
+      replication_task_id = "gitlab-db-clone-cdc"
+      migration_type      = "full-load-and-cdc"
+      source_endpoint_key = "source"
+      target_endpoint_key = "destination"
+      table_mappings = jsonencode({
+        "rules" : [
+          {
+            "rule-type" : "selection",
+            "rule-id" : "1",
+            "rule-name" : "select-all-tables",
+            "object-locator" : {
+              "schema-name" : "public",
+              "table-name" : "%"
+            },
+            "rule-action" : "include",
+            "filters" : []
+          },
+          # ########################################
+          # Drop sensitive columns from users tables
+          # ########################################
```

Comment:
What is the threat model driving this? Is there some security consideration with the analysis DB that doesn't apply to the real gitlab DB? Or are we just excluding this data to avoid storing a bunch of useless columns in the cloned DB?